examples/tv-casting-app/linux/main.cpp

```diff
@@ -21,18 +21,65 @@
 #include <controller/CHIPCommissionableNodeController.h>
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/ConfigurationManager.h>
+#include <support/CHIPArgParser.hpp>
 #include <system/SystemLayer.h>
 #include <transport/raw/PeerAddress.h>
 
 using namespace chip;
 using namespace chip::Controller;
+using chip::ArgParser::HelpOptions;
+using chip::ArgParser::OptionDef;
+using chip::ArgParser::OptionSet;
 
-constexpr int kTvDeviceType                         = 35;
-constexpr uint16_t commissioningWindowTimeoutInSec  = 3 * 60;
-constexpr uint32_t commissionerDiscoveryTimeoutInMs = 5 * 1000;
+constexpr int kVideoPlayerDeviceType                 = 35;
+constexpr uint16_t kOptionDeviceType                 = 't';
+constexpr uint16_t kCommissioningWindowTimeoutInSec  = 3 * 60;
+constexpr uint32_t kcommissionerDiscoveryTimeoutInMs = 5 * 1000;
 
-CommissionableNodeController commissionableNodeController;
-chip::System::SocketWatchToken token;
+CommissionableNodeController gCommissionableNodeController;
+chip::System::SocketWatchToken gToken;
+Mdns::DiscoveryFilter gDiscoveryFilter = Mdns::DiscoveryFilter(Mdns::DiscoveryFilterType::kDeviceType, kVideoPlayerDeviceType);
+
+bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue)
+{
+    bool retval = true;
+
+    switch (aIdentifier)
+    {
+    case kOptionDeviceType:
+        if (strcasecmp(aValue, "all") == 0)
+        {
+            gDiscoveryFilter = Mdns::DiscoveryFilter();
+        }
+        else
+        {
+            uint16_t deviceType = (uint16_t) strtol(aValue, nullptr, 10);
```

Comment:
could we use string parsing to convert to constant? saying `-t 35` is much more cryptic than `-t tv`. This could help us understand what device types are supported.

Reply:
Is there a way in the SDK to convert from a device type string to a device type integer? If not, I feel that a mapping util like that should be added in a separate PR - I can log an issue to track that.
This cmd line option basically allows the user to discover and do commissioning using any CHIP commissioner with a valid device type id.

Reply:
We won't have a complete list of device types until the device library gets finished so this additional work should probably wait until then.

Reply:
We should do it now and expand the list later - as we add more devices the list will grow here.
We can start witth supporting just "video".

or at least make string compare first to support video and then do the "parse as integer" bit. Allows flexibility but help users type strings and not numbers. Video is even 35, so not even a natural '1' or something.

Reply:
The list of devices does not need to be complete. A short (even very short ... mabe 1 more than tv) would make the code clearer. I would expect the code to do 'go through a list of known devicees at this time to make invocations easy' and as a fallback allow numbers.

Reply:
Updated the code to now accept text in addition to numeric representations of device type.

Reply:
Got a short list from Cam, some items that exist:
- 0x0000 OnOff switch
- 0x0001 Level control switch
- 0x0002 OnOff Output
- 0x0003 Level controllable output
- 0x0004 Scene selector
....


I would suggest the following for us to  use as examples
   0x0006 - remote control
   0x0022 - speaker
   0x0023 - Video Player

As these seemed more 'media types' somehow

Reply:
Resolving: we have at least 2 device types and list is extensible (we can add more as needed)